i6_models/primitives/feature_extraction.py

```diff
@@ -111,3 +117,80 @@ def forward(self, raw_audio, length) -> Tuple[torch.Tensor, torch.Tensor]:
             length = ((length - self.n_fft) // self.hop_length) + 1
 
         return feature_data, length.int()
+
+
+@dataclass
+class RasrCompatibleLogMelFeatureExtractionV1Config(ModelConfiguration):
+    """
+    Attributes:
+        sample_rate: audio sample rate in Hz
+        win_size: window size in seconds
+        hop_size: window shift in seconds
+        min_amp: minimum amplitude for safe log
+        num_filters: number of mel windows
+    """
+
+    sample_rate: int
+    win_size: float
+    hop_size: float
+    min_amp: float
+    num_filters: int
+
+    def __post_init__(self) -> None:
+        super().__post_init__()
+        assert self.win_size > 0 and self.hop_size > 0, "window settings need to be positive"
+        assert self.num_filters > 0, "number of filters needs to be positive"
+        assert self.hop_size <= self.win_size, "using a larger hop size than window size does not make sense"
+
+
+class RasrCompatibleLogMelFeatureExtractionV1(nn.Module):
+    """
+    Rasr-compatible log-mel feature extraction using log10. Does not use torchaudio.
+
+    Using it wrapped with torch.no_grad() is recommended if no gradient is needed
```

Comment:
I don't really understand this comment on `no_grad`. Isn't this just true in general for any PyTorch code? Why mention this explicitly here? Or is this module somehow special w.r.t. this?